SpiceSharp/Circuits/EntityCollection.cs

```diff
@@ -198,6 +232,38 @@ public void Remove(params string[] names)
             }
         }
 
+        /// <summary>
+        /// Removes the specified entities from the collection.
+        /// </summary>
+        /// <param name="entities">The entities.</param>
+        public void Remove(params Entity[] entities)
+        {
+            try
+            {
+                _lock.EnterWriteLock();
+                if (entities == null)
+                    return;
+                foreach (var entity in entities)
+                {
+                    if (_entities.ContainsKey(entity.Name))
+                    {
+                        _entities.Remove(entity.Name);
+                        _ordered.Remove(entity);
+                        OnEntityRemoved(new EntityEventArgs(entity));
+                    }
+                }
+            }
```

Comment:
I'm having a hard time thinking of a case where this would be useful/necessary. I won't put it in this update, but if it comes up somewhere I'll reconsider it.